tools/docker.sh

```diff
@@ -2,7 +2,37 @@
 
 set -e
 
-for tag in latest $(git tag --points-at | sed s/^v//); do
-  docker build -t raviqqe/muffet:$tag .
-  docker push raviqqe/muffet:$tag
-done
+check_buildx(){
+    if docker buildx version &>/dev/null; then
+        return 0
+    fi
+    if docker buildx ls &>/dev/null; then
+        return 0
+    fi
+    return 1
+}
+
+build_multiarch() {
+    if ! check_buildx; then
+        echo "Error: Docker buildx is not installed or not working"
+        return 1
+    fi
+
+    local platforms=${1:-"linux/amd64"}
+    
+    docker buildx create --name multiarch --use || true
+    
+    echo "Building for platforms: $platforms"
+    
+    # Build for each git tag and latest
+    for tag in latest $(git tag --points-at | sed s/^v//); do
+        docker buildx build \
+            --platform "$platforms" \
+            --tag raviqqe/muffet:$tag \
+            --push \
+            .
+    done
+}
+
+
+build_multiarch "$@"
```

Comment:
Not sure if building the image for multiple architectures as  the only option is the best solution.

Why don't you add  the regular build as the default behavior in case the user don't have `buildx` installed or they don't pass any platform as a parameter? 

```suggestion
build_image "$@"
```